programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -377,9 +377,16 @@ impl<'a, 'info> ScanningAccountRetriever<'a, 'info> {
         let n_perps = perp_index_map.len();
         let perp_oracles_start = perps_start + n_perps;
         let serum3_start = perp_oracles_start + n_perps;
-        let n_serum3 = ais[serum3_start..]
+        let mut n_serum3 = 0;
+        ais[serum3_start..]
             .iter()
-            .filter(|x| x.data_len() == std::mem::size_of::<serum_dex::state::OpenOrders>() + 12)
+            .inspect(|x| {
+                if x.data_len() == std::mem::size_of::<serum_dex::state::OpenOrders>() + 12
+                    && serum3_cpi::has_serum_header(&x.data.borrow())
+                {
+                    n_serum3 += 1;
+                }
+            })
             .count();
```

Comment:
I still think this is not quite right. What if the user passes "serum oo, oracle, another serum oo" - now n_serum3 would be 2 presumably? That's why I suggest using `slice.iter().position(|x|!is_serum(x)).unwrap_or(slice.len)`.

Or, to keep it similar to what's done for the previous, you could also do `.take_while(|x| is_serum(x)).count()`